frame/staking/src/pallet/impls.rs

```diff
@@ -871,12 +879,15 @@ impl<T: Config> Pallet<T> {
 	/// to `Nominators` or `VoterList` outside of this function is almost certainly
 	/// wrong.
 	pub fn do_add_nominator(who: &T::AccountId, nominations: Nominations<T>) {
-		if !Nominators::<T>::contains_key(who) {
+		let prev_nominations = Self::nominations(who);
+		if !prev_nominations.is_none() {
 			// maybe update sorted list.
 			let _ = T::VoterList::on_insert(who.clone(), Self::weight_of(who))
 				.defensive_unwrap_or_default();
 		}
 		Nominators::<T>::insert(who, nominations);
+		let _ = T::EventListener::on_nominator_add(who, prev_nominations.unwrap_or_default())
```

Comment:
I think the types here kinda show that `on_nominator_add` is actually being called both when a nominator is being *added*, and when it is being *updated*..

Reply:
Well, it's called within `do_add_nominator` so we are consistent with our naming. I don't think it makes much sense to have a name `on_nominator_updated` for a method that's only called in `do_add_nominator`.